pkg/endpoint/bpf.go

```diff
@@ -902,6 +902,96 @@ func (e *Endpoint) GetBPFValue() (*lxcmap.EndpointInfo, error) {
 	return info, nil
 }
 
+// syncPolicyMapDelta updates the bpf policy map state based on the
+// difference between the realized and desired policy state without
+// dumping the bpf policy map.
+func (e *Endpoint) syncPolicyMapDelta() error {
+	// Nothing to do if the desired policy is already fully realized.
+	if e.realizedPolicy == e.desiredPolicy {
+		return nil
+	}
+
+	errors := []error{}
+
+	// Delete policy keys present in the realized state, but not present in the desired state
+	for keyToDelete := range e.realizedPolicy.PolicyMapState {
+		// If key that is in realized state is not in desired state, just remove it.
+		if _, ok := e.desiredPolicy.PolicyMapState[keyToDelete]; !ok {
+
+			// Convert from policy.Key to policymap.Key
+			policyKeyToPolicyMapKey := policymap.PolicyKey{
+				Identity:         keyToDelete.Identity,
+				DestPort:         keyToDelete.DestPort,
+				Nexthdr:          keyToDelete.Nexthdr,
+				TrafficDirection: keyToDelete.TrafficDirection,
+			}
+
+			err := e.PolicyMap.DeleteKey(policyKeyToPolicyMapKey)
+			if err != nil {
+				// TODO: Maybe we should ignore delete errors
```

Comment:
I think that we should definitely keep it as is and log an error - it indicates that something is out of whack with the endpoint lifecycle.